### DISCUSSION TITLE
Async operations within `$extends`

### DISCUSSION THREAD

**Discussion by Rykuno** [prisma/prisma#17927] [Q&A]
Within the `clientExtentions` previewFeature is it possible to run async queries for the computed field value?

Example, for each `Post` I'd like to extend the model to know if a user `hasLiked`; or something of the sort. 

Something like the following. Is this possible or is there another better way using prisma?

```ts
      const xprisma = ctx.prisma.$extends({
        result: {
          post: {
            hasLiked: {
              needs: { id: true },
              async compute(data) {
                const hasLiked = await ctx.prisma.like.count({
                  where: { postId, userId }
                });
                return !!hasLiked;
              }
            }
          }
        }
      });
      ```

**Comment by nurul3101** [comment]
Hey @Rykuno 👋 

I believe passing an async function is possible. I tested it with this simple example:
```typescript
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient().$extends({
  result: {
    user: {
      getUser: {
        needs: {},
        compute() {
          return async () => {
            const res = await prisma.user.findUnique({
              where: {
                id: '1',
              },
            });

            return res;
          };
        },
      },
    },
  },
});

async function main() {
  const user = await prisma.user.create({
    data: {
      id: '1',
      firstName: 'John',
      lastName: 'Doe',
    },
  });

  const res = await user.getUser();

  console.log(JSON.stringify(res, null, 2));
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```

This script defines a `getUser` async function that gets user details.

Here's the response:
<img width="513" alt="Screenshot 2023-02-15 at 4 17 33 PM" src="https://user-images.githubusercontent.com/13049676/219006573-343e0b72-936d-4447-bafc-f64158244441.png">


```

**Reply by Rykuno** [reply]
@nurul3101 I'm looking to have it as a sort of `FieldResolver` on the `Post` field. So that each `Post` in a `findMany` will return an additional field called `hasLiked`, rather than a function. Unsure if this is possible with its current implementation but i hope I'm wrong.

**Reply by N0N4M3BNS** [reply]
@nurul3101 I try your method, but return null to me

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
